### ISSUE TITLE
Previews for gradients and images

### ISSUE THREAD

**Issue by mattpass** [icecoder/ICEcoder#111]
There are previews for flat colors but it would be great to also see this for gradients and images.

**Comment by demolitions** [comment]
Once you have a CSS rule written in the editor buffer, a simple JS hack should be sufficient to "copy" the CSS code from the editor buffer to a `.css_gradient_nn` rule in a `<style>` tag into the page's `<head>`, then apply that class to a little box in the gutter.
How do you like this idea?
jQuery could make this a bit easier.

**Comment by mattpass** [comment]
Having a box in the gutter sounds nice, tho different behaviour from the color picker box that appears when your cursor is on a color value.  So maybe best to stick with doing it the same way:

![image](https://user-images.githubusercontent.com/1311155/27228197-838e06d6-529e-11e7-8cec-785cee55318d.png)

We should really follow the same behaviour, especially as the gutter is already taken up by line numbers, fold marks and JS Hint errors if plugin installed.

So maybe if your cursor is on a gradient or image value, it displays a box to the bottom right of the cursor in the same way, showing the gradient, or an icon for the image. Clicking this will show a picker for gradient or the image viewer if an image (same as if you double click an image in file manager right now).

jQuery may make things easier, but it's not something I want in this repo due to filesize bloat.

**Comment by mattpass** [comment]
ie - for color gradients:

![image](https://user-images.githubusercontent.com/1311155/27228580-1773682c-52a0-11e7-8226-d1e5a38b44e5.png)

and for images:

![image](https://user-images.githubusercontent.com/1311155/27228605-3a833ba8-52a0-11e7-88a7-dabffba9b50a.png)

**Comment by demolitions** [comment]
You're right, I was confusing ICEcoder's color display with NetBeans' one.
The technique should however work if you put the css class on a tooltip.
The picker would be a bit harder to implement.

**Comment by mattpass** [comment]
Yeah, having them in the gutter does look nice, but we already have gutter things so maybe would become cluttered.

You could put it in a tooltip, but CodeMirror provides a 'widget' system where you can add arbitrary DOM elems to the editor - so, HTML, an image, whatever. Probably easy enough to find the code I have for flat color boxes right now and make it display a gradient box or an image icon.

The trickiest bit as you say is implementing a picker. I wouldn't personally build one as it'd take ages, so happy to have a lib for that provided it's solid and v small (just a few kbs).

For instance, the color picker for flat colors I used a lib called Farbtastic. Shame that doesn't do gradients. :-) If you can find a lib, I'd happily vet it and then can be implemented?

**Comment by demolitions** [comment]
IIRC I should already have an image-based gradient picker written in pure JS, shouldn't be too much of a hassle to modify it to use farbtastic and update it to CSS3.
I'll try to fetch it from one of my old servers and put it to good use.

**Comment by mattpass** [comment]
Nice! If we could use Farbtastic with it that'd be great. Look forward to seeing what you come up with,
